### PR TITLE
Material upload colors

### PR DESCRIPTION
**Pull request by Iryna-Bzdyr**
added  material-color editing services

### CODE REVIEW

File: modules/material/material.resolver.js [hunk 1]
```diff
@@ -16,6 +16,16 @@ const materialQuery = {
       message: MATERIAL_NOT_FOUND,
     };
   },
+  getMaterialColorByCode: async (parent, args) => {
+    const material = await materialService.getMaterialColorByCode(args.code);
+    if (material[0].colors[0]) {
+      return material[0].colors[0];
```
Comment (TorskyM) [inline modules/material/material.resolver.js:22]: Move this logic to service, we should try to remain resolvers as clean as possible

File: modules/material/material.resolver.js [hunk 2]
```diff
@@ -16,6 +16,16 @@ const materialQuery = {
...
+    }
+    return {
+      statusCode: 404,
+      message: COLOR_NOT_FOUND,
```
Comment (TorskyM) [inline modules/material/material.resolver.js:26]: Try to use the same approach to the implementation of the method, move error handling to services as you do in other methods

File: modules/material/material.resolver.js [hunk 3]
```diff
@@ -29,6 +39,23 @@ const materialMutation = {
       };
     }
   },
+
+  addMaterialColor: async (parent, args) => {
+    args.color.images = args.image;
```
Comment (TorskyM) [inline modules/material/material.resolver.js:44]: Why do you need this assignment?

File: modules/material/material.resolver.js (hunk 2 above)
Comment (Iryna-Bzdyr) [reply modules/material/material.resolver.js]: Fixed

File: modules/material/material.resolver.js (hunk 3 above)
Comment (Iryna-Bzdyr) [reply modules/material/material.resolver.js]: Fixed

Comment (Iryna-Bzdyr) [reply modules/material/material.resolver.js]: Fixed

File: modules/material/material.resolver.js (hunk 1 above)
Comment (Iryna-Bzdyr) [reply modules/material/material.resolver.js]: Fixed

File: typeDefs.js [hunk 4]
```diff
@@ -447,13 +455,19 @@ const typeDefs = gql`
...
       material: MaterialInput!
-      images: Upload!
+      images: Upload
     ): MaterialResult
+    addMaterialColor(
+      id: ID
+      color: MaterialColorInput
+      image: Upload
+    ): MaterialColorResult
+    deleteMaterialColor(id: ID, code: Int): MaterialResult
```
Comment (TorskyM) [inline typeDefs.js:470]: The ID shouldn't be nullable

File: typeDefs.js [hunk 5]
```diff
@@ -447,13 +455,19 @@ const typeDefs = gql`
...
+    addMaterial(material: MaterialInput!, images: Upload!): MaterialResult
     deleteMaterial(id: ID!): MaterialResult
     updateMaterial(
       id: ID!
       material: MaterialInput!
-      images: Upload!
+      images: Upload
     ): MaterialResult
+    addMaterialColor(
+      id: ID
```
Comment (TorskyM) [inline typeDefs.js:466]: The ID shouldn't be nullable

**Review by ahd3r — APPROVED** [review]

File: typeDefs.js (hunk 4 above)
Comment (Iryna-Bzdyr) [reply typeDefs.js]: Fixed

File: typeDefs.js (hunk 5 above)
Comment (Iryna-Bzdyr) [reply typeDefs.js]: Fixed

**Review by ecl1pseo — APPROVED** [review]

**Review by TorskyM — APPROVED** [review]

**Pull request merged**